src/mvhistory.jl

```diff
@@ -97,3 +99,16 @@ macro trace(tr, i, vars...)
     end
     block
 end
+
+"""
+    pushoradd!(trace, key, iter, val)
+
+Increments the value for a given key and iteration if it exists, otherwise adds the key/iteration pair with an ordinary push.
+"""
+function pushoradd!(trace::MultivalueHistory, key::Symbol, iter::Number, val)
+    if haskey(trace, key) && length(trace.storage[key]) >= iter
```

Comment:
`length(trace.storage[key]) >= iter`

i don't think that works in the general case. One can store disconnected iterations that don't follow an explicit system other than being ordered (e.g. the iterations `1, 10, 12, 100`).  So i think to find the actual index of the associated value of `iter` one would have to use something like 

```julia
i = findfirst(trace.storage[key].iterations, iter)
```

with that `i` one can then check if `i !=0 ` (i.e. the element exists) and if it does use 

```julia
trace[key].values[i] += val
```